### PR TITLE
Run hls::cne and llvm::dne as part of the passes

### PR DESCRIPTION
**Pull request by sjalander**

### CODE REVIEW

File: jlm/hls/backend/rvsdg2rhls/add-forks.cpp [hunk 1]
```diff
@@ -37,8 +37,8 @@ add_forks(jlm::rvsdg::region * region)
         add_forks(structnode->subregion(n));
       }
     }
-    // If a node has no inputs it is a constant
-    bool isConstant = node->ninputs() == 0;
+    bool isConstant = rvsdg::is<rvsdg::bitconstant_op>(node->operation())
+                   || rvsdg::is<rvsdg::ctlconstant_op>(node->operation());
```
Comment (phate) [inline jlm/hls/backend/rvsdg2rhls/add-forks.cpp:41]: What is the reason for changing this?

**Review by phate — DISMISSED** [review]

File: jlm/hls/backend/rvsdg2rhls/add-forks.cpp (hunk 1 above)
Comment (phate) [reply jlm/hls/backend/rvsdg2rhls/add-forks.cpp:41]: @sjalander

Comment (sjalander) [reply jlm/hls/backend/rvsdg2rhls/add-forks.cpp:41]: @davidmetz 
Can you explain why this change has been made?

This is how the code is for the addr-queue branch, but the earlier version of the code, which checks for no inputs, also works for the benchmarks that are included in the hls-test-suite.

Comment (phate) [reply jlm/hls/backend/rvsdg2rhls/add-forks.cpp:41]: @sjalander @davidmetz

Comment (sjalander) [reply jlm/hls/backend/rvsdg2rhls/add-forks.cpp:41]: @phate 
I discussed with David, and this change was to handle local memory. But further changes has med this unnecessary as the test suite works without it.
So I've reverted the change.

**Review by phate — APPROVED** [review]

**Pull request merged**